swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/snapshots/AmazonSnapshotHandler.kt

```diff
@@ -0,0 +1,151 @@
+  exclusionPolicies: List<ResourceExclusionPolicy>,
+  applicationEventPublisher: ApplicationEventPublisher,
+  lockingService: Optional<LockingService>,
+  retrySupport: RetrySupport,
+  dynamicConfigService: DynamicConfigService,
+  private val rules: List<Rule<AmazonSnapshot>>,
+  private val snapshotProvider: ResourceProvider<AmazonSnapshot>,
+  private val orcaService: OrcaService,
+  private val applicationUtils: ApplicationUtils,
+  private val taskTrackingRepository: TaskTrackingRepository,
+  private val resourceUseTrackingRepository: ResourceUseTrackingRepository,
+  private val usedSnapshotRepository: UsedSnapshotRepository,
+  private val swabbieProperties: SwabbieProperties
+) : AbstractResourceTypeHandler<AmazonSnapshot>(
+  registry,
+  clock,
+  rules,
+  resourceTrackingRepository,
+  resourceStateRepository,
+  exclusionPolicies,
+  resourceOwnerResolver,
+  notifiers,
+  applicationEventPublisher,
+  lockingService,
+  retrySupport,
+  resourceUseTrackingRepository,
+  swabbieProperties,
+  dynamicConfigService
+) {
+  override fun deleteResources(markedResources: List<MarkedResource>, workConfiguration: WorkConfiguration) {
+    TODO("not implemented")
+  }
+
+  override fun handles(workConfiguration: WorkConfiguration): Boolean =
+    workConfiguration.resourceType == SNAPSHOT && workConfiguration.cloudProvider == AWS && !rules.isEmpty()
+
+  override fun getCandidate(resourceId: String, resourceName: String, workConfiguration: WorkConfiguration): AmazonSnapshot? {
+    val params = Parameters(
+      id = resourceId,
+      account = workConfiguration.account.accountId!!,
+      region = workConfiguration.location,
+      environment = workConfiguration.account.environment
+    )
+
+    return snapshotProvider.getOne(params)
+  }
+
+  override fun getCandidates(workConfiguration: WorkConfiguration): List<AmazonSnapshot>? {
+    val params = Parameters(
+      account = workConfiguration.account.accountId!!,
+      region = workConfiguration.location,
+      environment = workConfiguration.account.environment
+    )
+
+    return snapshotProvider.getAll(params).also { snapshots ->
+      log.info("Got {} snapshots.", snapshots?.size)
+    }
+  }
+
+  override fun preProcessCandidates(
+    candidates: List<AmazonSnapshot>,
+    workConfiguration: WorkConfiguration
+  ): List<AmazonSnapshot> {
+    checkIfImagesExist(
+      candidates = candidates,
+      params = Parameters(
+        account = workConfiguration.account.accountId!!,
+        region = workConfiguration.location,
+        environment = workConfiguration.account.environment
+      ))
+    return candidates
+  }
+
+  private fun checkIfImagesExist(candidates: List<AmazonSnapshot>, params: Parameters) {
+    log.info("Checking for existing images for {} snapshots. Params: {} ", candidates.size, params)
+    candidates.forEach { snapshot ->
+      if (snapshot.description == null) {
+        snapshot.set(SNAPSHOT_NOT_FROM_BAKE, true)
```

Comment:
Alternatively, we can use the generic `NAIVE_EXCLUSION` classifier here. This avoids tying to a netflix only pattern.

Reply:
Can you explain more? What general set of things does `NAIVE_EXCLUSION` capture?

Reply:
`NAIVE_EXCLUSION` is a catch-all exclusion classifier that would make the resource excludable by default.